### ISSUE TITLE
every epoch is giving nan value while training

### ISSUE THREAD

**Issue by Berrlinn** [MaybeShewill-CV/lanenet-lane-detection#536]
I tried running this script 'python tools/train_lanenet_tusimple.py' in Google colab

Output: 
2022-01-02 16:05:02.906 | INFO     | trainner.tusimple_lanenet_multi_gpu_trainner:train:397 - => Starts to train LaneNet from scratch ...
0it [00:00, ?it/s]
W0102 16:05:02.908667 4984 warnings.py:110] /usr/local/lib/python3.7/dist-packages/numpy/core/fromnumeric.py:3118: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)

W0102 16:05:02.909043 4984 warnings.py:110] /usr/local/lib/python3.7/dist-packages/numpy/core/_methods.py:85: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)

0it [00:00, ?it/s]
2022-01-02 16:05:02.909 | INFO     | trainner.tusimple_lanenet_multi_gpu_trainner:train:531 - => Epoch: 1 Time: 2022-01-02 16:05:02 Train loss: nan Train miou: nan Val loss: nan Val miou: nan...
0it [00:00, ?it/s]
0it [00:00, ?it/s]
2022-01-02 16:05:02.911 | INFO     | trainner.tusimple_lanenet_multi_gpu_trainner:train:531 - => Epoch: 2 Time: 2022-01-02 16:05:02 Train loss: nan Train miou: nan Val loss: nan Val miou: nan...
.
.
.
2022-01-02 16:05:19.438 | INFO     | trainner.tusimple_lanenet_multi_gpu_trainner:train:531 - => Epoch: 900 Time: 2022-01-02 16:05:19 Train loss: nan Train miou: nan Val loss: nan Val miou: nan...
0it [00:00, ?it/s]
0it [00:00, ?it/s]
2022-01-02 16:05:19.439 | INFO     | trainner.tusimple_lanenet_multi_gpu_trainner:train:531 - => Epoch: 901 Time: 2022-01-02 16:05:19 Train loss: nan Train miou: nan Val loss: nan Val miou: nan...
0it [00:00, ?it/s]
0it [00:00, ?it/s]
2022-01-02 16:05:19.441 | INFO     | trainner.tusimple_lanenet_multi_gpu_trainner:train:531 - => Epoch: 902 Time: 2022-01-02 16:05:19 Train loss: nan Train miou: nan Val loss: nan Val miou: nan...
0it [00:00, ?it/s]
0it [00:00, ?it/s]
2022-01-02 16:05:19.443 | INFO     | trainner.tusimple_lanenet_multi_gpu_trainner:train:531 - => Epoch: 903 Time: 2022-01-02 16:05:19 Train loss: nan Train miou: nan Val loss: nan Val miou: nan...
0it [00:00, ?it/s]
0it [00:00, ?it/s]
2022-01-02 16:05:19.445 | INFO     | trainner.tusimple_lanenet_multi_gpu_trainner:train:531 - => Epoch: 904 Time: 2022-01-02 16:05:19 Train loss: nan Train miou: nan Val loss: nan Val miou: nan...
2022-01-02 16:05:19.445 | INFO     | trainner.tusimple_lanenet_multi_gpu_trainner:train:543 - Best model's val mious are: [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan]
2022-01-02 16:05:19.445 | INFO     | trainner.tusimple_lanenet_multi_gpu_trainner:train:544 - Complete training process good luck!!

**Comment by ilpapds** [comment]
I experience the same issue. Did you find any solution to this problem? 
Thank you in advance

**Comment by diazoangga** [comment]
@ilpapds @Berrlinn Hey, good day guys, could you give me more detailed info on what dataset did you train on and what BATCH_SIZE did you set on the config file? (config file; config/tusimple_lanenet.yaml)

I experienced this also, so checked the codes in the 'data_provider/lanenet_data_feed_pipeline.py' file, and what i found was there is a syntax: "dataset.batch(batch_size=batch_size, drop_remainder=True)", which basically drops batches that has data less than BATCH_SIZE when the codes trying to batch the dataset.

so, if you're using the example dataset that comes with this repository (located in data/training_data_examples), which only has 5 images, then you might have to change the BATCH_SIZE less than 5, maybe 2 even, just to test the codes out.

Hope it answers your question!
Let me know the updates

**Comment by 12lvxiaofeng12** [comment]
我在训练自己的数据集时出现了一样的情况（出现nan），请问有什么解决方法吗，希望您能回答我的问题，谢谢!

**Comment by 12lvxiaofeng12** [comment]
我在训练自己的数据时出现了同样的情况（出现nan），请问有什么解决方法吗，希望您能回答我的问题，谢谢!

**Comment by 12lvxiaofeng12** [comment]
我在训练自己的数据时出现了同样的情况Train loss: nan Train miou: nan Val loss: nan Val miou: nan...，请问有什么解决方法吗，希望您能回答我的问题，谢谢

**Comment by 12lvxiaofeng12** [comment]
我在训练自己的数据时出现了同样的情况 Train loss: nan Train miou: nan Val loss: nan Val miou: nan...您有什么解决方法吗，希望能回答我的问题,谢谢！

**Comment by MaybeShewill-CV** [comment]
@Berrlinn If you use data in ./data/training_data_example try @diazoangga 's suggestions. If you use your own dataset there must be something wrong with your label:)

**Comment by MaybeShewill-CV** [comment]
@Berrlinn Check if your label image was correct:)

**Comment by Berrlinn** [comment]
Would you please explain what exactly I have to do?

**Comment by MaybeShewill-CV** [comment]
@Berrlinn use [0, 255] to label binary label image, use [0, 1, 2, 3, 4] to label instance label image if you have 5 lanes in that image:)

**Comment by Berrlinn** [comment]
I am new to all these. Can you tell which file should I update with these
changes?
Thanks.

**Comment by MaybeShewill-CV** [comment]
@Berrlinn Do not have enough time to explain all the details. You may find useful tips in closed issues:)
